Sources/Subs-Boards.php

```diff
@@ -664,45 +671,22 @@ function modifyBoard($board_id, &$boardOptions)
 			))
 		);
 
-	if (!empty($boardOptions['deny_groups']) || !empty($boardOptions['access_groups'])) {
-		// Before we add new access_groups or deny_groups, remove all of the old entries
-		$smcFunc['db_query']('', '
-			DELETE FROM {db_prefix}board_permissions_view
-			WHERE id_board = {int:selected_board}',
-			array(
-				'selected_board' => $board_id,
-			)
-		);
-	}
-
-	// Do permission sync
-	if (!empty($boardOptions['deny_groups']))
-	{
-		$insert = array();
-		foreach ($boardOptions['deny_groups'] as $value)
-			$insert[] = array($value, $board_id, 1);
-
-		$smcFunc['db_insert']('insert',
-			'{db_prefix}board_permissions_view',
-			array('id_group' => 'int', 'id_board' => 'int', 'deny' => 'int'),
-			$insert,
-			array('id_group', 'id_board', 'deny')
-		);
-	}
-
-	if (!empty($boardOptions['access_groups']))
-	{
-		$insert = array();
-		foreach ($boardOptions['access_groups'] as $value)
-			$insert[] = array($value, $board_id, 0);
+	// Before we add new access_groups or deny_groups, remove all of the old entries
+	$smcFunc['db_query']('', '
+		DELETE FROM {db_prefix}board_permissions_view
+		WHERE id_board = {int:selected_board}',
+		array(
+			'selected_board' => $board_id,
+		)
+	);
 
+	if ($board_permissions_inserts != [])
```

Comment:
Nit: SMF coding style prefers the `array()` syntax.
```suggestion
	if ($board_permissions_inserts != array())
```